hilti/toolchain/include/compiler/validator.h

```diff
@@ -38,6 +38,10 @@ class VisitorMixIn {
     /** Returns the number of errors reported so far. */
     auto errors() const { return _errors; }
 
+    /** Validates if provided type arguments match a type's expectation. */
+    void checkTypeArguments(const node::Range<Expression>& have, const node::Set<type::function::Parameter>& want,
+                            Node* n, bool allow_no_arguments = false, bool do_not_check_types = false);
```

Comment:
Nit: I'd rename these to get rid of negatives, e.g., here and also in the impl.

```suggestion
                            Node* n, bool allow_empty_args = false, bool check_types = true);
```

The `allow_empty_args` does not change semantics, but `do_not_check_types` does and would require flipping the callside and the check in the impl.

Reply:
I don't know but I prefer the negative versions to specify the "unusual" case.